### ISSUE TITLE
Required flag derived from NotEmpty annotation does not respect validation group of annotation

### ISSUE THREAD

**Issue by beranradek** [beranradek/formio#8]
Required flag on some form field is still true even if field with NonEmpty annotation has unused validation group specified.

**Comment by lesinsa** [comment]
Method net.formio.AbstractFormElement#isRequiredByAnnotations doesn't regards validation groups at all. Moreover, @Size annotation is handled incorrect
```
						Size s = (Size) ann;
						if (s.min() > 0) {
							required = true;
							break;
						}
```
Minimal size should be effective when @NotNull annotation is present only (recommended in JSR-303).

**Comment by lesinsa** [comment]
Is project alive?

**Comment by beranradek** [comment]
@lesinsa you are right that Size annotation presence should not be checked within isRequiredByAnnotations. Null elements should be valid when the Size annotation is specified. I have released version 1.6.4 with that fix.

Recognizing validation groups within isRequiredByAnnotations is more conceptual problem - validation groups passed into bind method can be propagated (and stored) at AbstractFormElement level. You are welcome to create a pull request if you want to implement this feature yourself. For now, the library evaluates annotations in static way regarding the isRequired() getter.
